arch/sim/src/sim/up_simsmp.c

```diff
@@ -120,7 +120,7 @@ void sched_note_cpu_resume(struct tcb_s *tcb, int cpu);
 static void *sim_idle_trampoline(void *arg)
 {
   struct sim_cpuinfo_s *cpuinfo = (struct sim_cpuinfo_s *)arg;
-#ifdef CONFIG_SIM_WALLTIME
+#ifdef CONFIG_SIM_WALLTIME_HOSTSLEEP
```

Comment:
don't need change, we need sleep in both case(sleep and signal). Let's revert the change at:
https://github.com/apache/incubator-nuttx/pull/1859/files#diff-d607b8ec5dc28fb1e92e2b4389e2a9a8R106

Reply:
Ok, I will revert this. Out of topic: I also want to integrate the SIGUSR1 to use the NuttX irq handling logic in another patch/PR.

Reply:
It's on my TODO list too, it's great that you have time to fix it.